src/components/global/UploadButton.vue

```diff
@@ -58,7 +63,33 @@ export default class UploadButton extends Props {
     let file_input = this.$refs['real-input'];
     (file_input as any).click();
   }
+
+  uploadFileDrop(e: DragEvent) {
+    if (e == null) return;
+    let droppedFiles = e.dataTransfer!.files;
+    let files = fix_files(droppedFiles);
+    if (!files) return;
+    if (files.length > 0) {
+      if (
+        // were all files sent in a supported format?
+        files.filter(file => file.type == 'application/json').length ==
+        files.length
+      ) {
+        // Notify we got files
+        this.$emit('files-selected', files);
+      } else {
+        return this.$toasted.global.error({
+          message: String(
+            'One or more files provided is not in a support format.  Please ' +
+              'upload file(s) in a format supported by Heimdall Lite (i.e., JSON)'
+          ),
+          isDark: this.$vuetify.theme.dark
+        });
+      }
+    }
+  }
```

Comment:
Is it possible for us to just leverage the way that a normal file upload works and have unified functionality whether the user drag and drops a file or clicks to upload a file?